src/com/google/javascript/jscomp/GatherModuleMetadata.java

```diff
@@ -233,6 +235,13 @@ private void visitImportOrExport(NodeTraversal t, Node importOrExport) {
       }
     }
 
+    private void visitDynamicImport(NodeTraversal t, Node dynamicImport) {
+      currentModule
+          .metadataBuilder
+          .es6ImportSpecifiersBuilder()
+          .add("*");
```

Comment:
Shouldn't this be looking at the children of the dynamic import node?

Reply:
I don't understand the module metadata completely, but these both refer to the same thing:

```js
import * as foo from './foo.js';
import('./foo.js').then((foo2) => {
   console.log(foo === foo2);
});
```

My logic was that a dynamic import was a "*" import so this was simply indicating that the currentModule referred to the module export namespace.